tests/test_utils_gz.py

```diff
@@ -27,3 +27,9 @@ def test_gunzip_truncated_short(self):
         with open(join(SAMPLEDIR, 'truncated-crc-error-short.gz'), 'rb') as f:
             text = gunzip(f.read())
             assert text.endswith(b'</html>')
+
+    def test_gunzip_illegal_eof(self):
+        with open(join(SAMPLEDIR, 'unexpected-eof.gz'), 'rb') as f:
+            text = gunzip(f.read())
+            with open(join(SAMPLEDIR, 'unexpected-eof-output.txt'), 'rb') as o:
+                self.assertEqual(len(text), len(o.read().decode("utf-8")))
```

Comment:
@jdemaeyer , the encodings of both text and the output of o.read() were different, in the new test I committed both are set to unicode, I added an assertion to make sure the content too is same.